### PR TITLE
FIX: fixing spmd tests utilities for dpctl inputs

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description

DPCTL doesn't support some numpy-like ufunc  functionality, For example, loop of ufunc does not support argument 0 of type dpctl.tensor._usmarray.usm_ndarray which has no callable conjugate method.

Make sense before some manipulations with resulted dpctl or any other non-numpy arrays convert to numpy before and do comparison with expected numpy arrays.

Found on #1861 

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [X] I have reviewed my changes thoroughly before submitting this pull request.
- [] I have commented my code, particularly in hard-to-understand areas.
- [] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [X] I have added a respective label(s) to PR if I have a permission for that.  
- [X] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [X] The unit tests pass successfully.
- [X] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: sklearnex/tests/_utils_spmd.py [hunk 1]
```diff
@@ -179,7 +182,11 @@ def _assert_kmeans_labels_allclose(
         AssertionError: If clusters are not correctly assigned.
     """
 
+    spmd_labels_as_numpy = _as_numpy(spmd_labels)
+    spmd_centers_as_numpy = _as_numpy(spmd_centers)
```
Comment (icfaust) [inline sklearnex/tests/_utils_spmd.py:186]:
```suggestion
    np_spmd_labels = _as_numpy(spmd_labels)
    np_spmd_centers = _as_numpy(spmd_centers)
```
Shortening without lost of information

File: sklearnex/tests/_utils_spmd.py [hunk 2]
```diff
@@ -146,8 +146,11 @@ def _assert_unordered_allclose(spmd_result, batch_result, localize=False, **kwar
     Raises:
         AssertionError: If results do not match.
     """
+    spmd_result_as_numpy = _as_numpy(spmd_result)
```
Comment (icfaust) [inline sklearnex/tests/_utils_spmd.py:149]:
```suggestion
    np_spmd_result = _as_numpy(spmd_result)
```

**Review by icfaust — COMMENTED** [review]
Small change in naming convention to keep it compact. Otherwise good to go.

**Review by olegkkruglov — APPROVED** [review]
LGTM, please make sure the naming dispute is settled

**Review by icfaust — APPROVED** [review]
I approve.  Just for posterity, and alternate solution as we have talked about elsewhere would have been to use get_namespace.  This may be a better convention going forward than using parametrized numpy dtypes, ```assert(data.dtype == getattr(get_namespace(data)[0], dtype.__name__))```

**Comment by samir-nasibli** [comment]
> I approve. Just for posterity, and alternate solution as we have talked about elsewhere would have been to use get_namespace. This may be a better convention going forward than using parametrized numpy dtypes, `assert(data.dtype == getattr(get_namespace(data)[0], dtype.__name__))`

I think this less common for our primitives, so better to convert to one dataframe format then compare.

**Comment by samir-nasibli** [comment]
@mergify backport rls/2024.7.0-rls

**Pull request merged**